### ISSUE TITLE
setLimit(0) on query returns no rows

### ISSUE THREAD

**Issue by dieseldjango** [couchbase/couchbase-lite-java-core#1625]
According to the API docs, "limit: If nonzero, this is the maximum number of rows that will be returned."

This gave me the impression that setting it to zero does not limit the results. However, this is not the case. It must be set to Integer.MAX_VALUE to have no limit. It would be nice to fix the code to impose no limit if the value is zero, or update the docs to be more clear.

**Comment by natgross** [comment]
If limit is NULL, the resultset is unlimited.

**Comment by dieseldjango** [comment]
That would be a great thing to put in the doc!

**Comment by hideki** [comment]
From doc:
```
Gets or sets the maximum number of rows to return. The default value is 0, meaning 'unlimited'.
```
https://developer.couchbase.com/documentation/mobile/1.4/references/couchbase-lite/couchbase-lite/query/query/index.html

This could be a bug.

**Comment by natgross** [comment]
@hideki Once upon a time it used to be that way. (0 meant unlimited.) Then after an update (don't recall which one), 0 actually meant zero! So, I figured "hey; that makes sense" when I LIMIT something to ZERO, it should be 0. (Unlike a "-1" which is more like a flag. )  So, I tried null, and yes(!), null meant unlimited. (Which also makes sense.) 
I would say that you have _a documentation error_  at worst.  So maybe just change the doc to read:

> Gets or sets the maximum number of rows to return. The default is null, meaning 'unlimited'.
OR
> Gets or sets the maximum number of rows to return. If null, it is 'unlimited'.

**Comment by natgross** [comment]
@hideki 
@dieseldjango
Sorry! for the confusion! I had built my own Query object to encapsulate cbl's. And used null as a flag in my api due to the fact that cbl's changed. In fact this is a line in my code with the my own comment that I had bumped into this problem.
_if (limit!=null && limit > 0) query.setLimit(limit); //don't even set limit to 0/-1/etc. these flags don't always work._
Anyhow, don't want to make this into a federal case. But since I already opened my big fat mouth, I am posting this comment as well, to round out the discussion (and not deleting the previous post.)
